tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -583,66 +582,12 @@ ZTEST(spi_loopback, test_spi_loopback)
 					  &async_evt, &caller, NULL,
 					  K_PRIO_COOP(7), 0, K_NO_WAIT);
 #endif
-	zassert_true(spi_is_ready_dt(&spi_slow), "Slow spi lookback device is not ready");
-
-	LOG_INF("SPI test slow config");
-
-	if (spi_complete_multiple(&spi_slow) ||
-	    spi_complete_loop(&spi_slow) ||
-	    spi_null_tx_buf(&spi_slow) ||
-	    spi_rx_half_start(&spi_slow) ||
-	    spi_rx_half_end(&spi_slow) ||
-	    spi_rx_every_4(&spi_slow) ||
-	    spi_rx_bigger_than_tx(&spi_slow) ||
-	    spi_complete_large_transfers(&spi_slow)
-#if (CONFIG_SPI_ASYNC)
-	    || spi_async_call(&spi_slow)
-#endif
-	    ) {
-		goto end;
-	}
-
-	zassert_true(spi_is_ready_dt(&spi_fast), "Fast spi lookback device is not ready");
-
-	LOG_INF("SPI test fast config");
-
-	if (spi_complete_multiple(&spi_fast) ||
-	    spi_complete_loop(&spi_fast) ||
-	    spi_null_tx_buf(&spi_fast) ||
-	    spi_rx_half_start(&spi_fast) ||
-	    spi_rx_half_end(&spi_fast) ||
-	    spi_rx_every_4(&spi_fast) ||
-	    spi_rx_bigger_than_tx(&spi_fast) ||
-	    spi_complete_large_transfers(&spi_fast)
-#if (CONFIG_SPI_ASYNC)
-	    || spi_async_call(&spi_fast)
-#endif
-	    ) {
-		goto end;
-	}
 
-	if (spi_resource_lock_test(&spi_slow, &spi_fast)) {
-		goto end;
-	}
+	ztest_run_all(NULL, false, ARRAY_SIZE(loopback_specs), 1);
 
-	LOG_INF("All tx/rx passed");
-end:
 #if (CONFIG_SPI_ASYNC)
 	k_thread_abort(async_thread_id);
-#else
-	;
 #endif
-}
 
-static void *spi_loopback_setup(void)
-{
-	memset(buffer_tx, 0, sizeof(buffer_tx));
-	memcpy(buffer_tx, tx_data, sizeof(tx_data));
-	memset(buffer2_tx, 0, sizeof(buffer2_tx));
-	memcpy(buffer2_tx, tx2_data, sizeof(tx2_data));
-	memset(large_buffer_tx, 0, sizeof(large_buffer_tx));
-	memcpy(large_buffer_tx, large_tx_data, sizeof(large_tx_data));
-	return NULL;
+	ztest_verify_all_test_suites_ran();
 }
-
-ZTEST_SUITE(spi_loopback, NULL, spi_loopback_setup, NULL, NULL, NULL);
```

Comment:
Is this not required to have this macro?
`ZTEST_SUITE(spi_loopback, NULL, NULL, NULL, NULL, NULL);`

Reply:
Not sure what you mean by this comment